pkg/azure/metrics/metrics.go

```diff
@@ -0,0 +1,291 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package metrics
+
+import (
+	"fmt"
+	"net/http"
+	"sort"
+	"strconv"
+	"strings"
+
+	operatorMetrics "github.com/cilium/cilium/operator/metrics"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	// See https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/request-limits-and-throttling#remaining-requests
+	// tags are conditionally set based on description
+	metricAzureRateLimitRemaining = prometheus.NewGaugeVec(prometheus.GaugeOpts{
```

Comment:
The current approach to declaring new metrics is using the `metric.Metric` cell type. For example: https://github.com/cilium/cilium/blob/24837c0339e0ec0b75e8eb2715921b2e6b0f5661/operator/identitygc/metrics.go#L42-L73

Could we create this via that method and have them be injected into the azure ipam allocator Cell?